dttools/src/bucketing_manager.c

```diff
@@ -52,23 +68,51 @@ void bucketing_manager_delete(bucketing_manager_t* m)
     free(m);
 }
 
-void bucketing_manager_add_resource_type(bucketing_manager_t* m, const char* r, double default_value, int num_sampling_points, double increase_rate, int max_num_buckets)
+void bucketing_manager_add_resource_type(bucketing_manager_t* m, const char* r, 
+        int set_default, double default_value, int num_sampling_points, 
+        double increase_rate, int max_num_buckets, int update_epoch)
 {
     if (!m)
     {
         fatal("No bucketing manager to add resource\n");
         return;
     }
 
-    /* only add resource type if it doesn't exist, do nothing otherwise */
+    bucketing_state_t* b; 
+    /* only add resource type if it doesn't exist, warn otherwise */
     if (!hash_table_lookup(m->res_type_to_bucketing_state, r))
     {
-        bucketing_state_t* b = bucketing_state_create(default_value, num_sampling_points, increase_rate, max_num_buckets, m->mode);
+        if (set_default)
+        {
+            if (!strncmp(r, "cores", strlen("cores")))
+            {
+                b = bucketing_state_create(1, 10, 2, 10, m->mode, 1);
```

Comment:
These parameters should be named, maybe as const at the top of the file, like `bucket_default_....`